Prog/Housing market analysis.sas

```diff
@@ -0,0 +1,414 @@
+		6="Arlington"
+		7="Fairfax, Fairfax city and Falls Church"
+		8="Loudoun"
+		9="Prince William, Manassas and Manassas Park"
+		10="Alexandria"
+		11="Total"
+		;
+
+	value structure
+		1= "Single family"
+		2= "Duplex"
+		3= "Small multifamily"
+		4= "Large multifamily"
+		.n= "Other"
+		 ;
+run;
+
+
+/**************************************************************************
+Part 1: Compile housing units by characteristics from Ipums.
+**************************************************************************/
+
+%macro COGunits(year);
+
+/* Calculate vacant untis for each year of Ipums */
+data COGSvacant_&year.(where=(vacancy in (1,2)));
+
+	%if &year. = 2017 %then %do;
+	set Ipums.Acs_&year._vacant_dc Ipums.Acs_&year._vacant_md Ipums.Acs_&year._vacant_va ;
+		%newpuma_jurisdiction; 
+		if upuma in (&pumanew.);
+	%end;
+	%else %if &year. = 2010 %then %do;
+	set Ipums.Acs_&year._vacant_dc Ipums.Acs_&year._vacant_md Ipums.Acs_&year._vacant_va ;
+		%oldpuma_jurisdiction; 
+		if upuma in (&pumaold.);
+	%end; 
+	%else %if &year. = 2000 %then %do;
+	set Ipums.Ipums_2000_vacant_dc Ipums.Ipums_2000_vacant_md Ipums.Ipums_2000_vacant_va ;
+		%oldpuma_jurisdiction; 
+		if upuma in (&pumaold.);
+	%end; 
+
+	if UNITSSTR in (03, 04) then structuretype=1; /*single family*/
+	if UNITSSTR =05 then structuretype=2; /*duplex*/
+	if UNITSSTR in (06, 07) then structuretype=3; /*small multifamily*/
+	if UNITSSTR in (08, 09, 10)then structuretype=4; /*large multifamily*/
+
+	if vacancy=1 then Tenure = 1; /*renter*/
+	if vacancy=2 then Tenure = 2; /*owner*/
+
+	vacantunit_&year.=1;
+
+run;
+
+proc summary data= COGSvacant_&year.;
+	class Jurisdiction structuretype bedrooms Tenure;
+	var vacantunit_&year.;
+	weight hhwt;
+	output out= COGSvacantunits_&year. sum=;
+run;
+
+proc sort data= COGSvacantunits_&year.;
+	by Jurisdiction structuretype bedrooms Tenure _TYPE_;
+run;
+
+/* Calculate total number of units for each year of Ipums */
+data COGSarea_&year. (where=(pernum=1 and gq in (1,2) and ownershpd in ( 12,13,21,22 )));
+	%if &year. = 2017 %then %do;
+	set Ipums.Acs_&year._dc Ipums.Acs_&year._md Ipums.Acs_&year._va; 
+		%newpuma_jurisdiction; 
+		if upuma in (&pumanew.);
+	%end;
+	%else %if &year. = 2010 %then %do;
+	set Ipums.Acs_&year._dc Ipums.Acs_&year._md Ipums.Acs_&year._va;
+		%oldpuma_jurisdiction; 
+		if upuma in (&pumaold.);
+	%end; 
+	%else %if &year. = 2000 %then %do;
+	set Ipums.Ipums_2000_dc Ipums.Ipums_2000_md Ipums.Ipums_2000_va;
+		%oldpuma_jurisdiction; 
+		ownershpd = ownershd;
+		if upuma in (&pumaold.);
+	%end; 
+
+	if UNITSSTR in (03, 04) then structuretype=1; /*single family*/
+	else if UNITSSTR = 05 then structuretype=2; /*duplex*/
+	else if UNITSSTR in (06, 07) then structuretype=3; /*small multifamily*/
+	else if UNITSSTR in (08, 09, 10)then structuretype=4; /*large multifamily*/
+
+
+	if ownershpd in (21, 22) then Tenure = 1; /*renter*/
+	else if ownershpd in ( 12,13 ) then Tenure = 2; /*owner*/
+
+	unit_&year.=1;
```

Comment:
I moved this into the final data step and renamed it totunits_&year. - I think having a count of all units (vacant and occupied) is more useful, since the data already has a vacancy flag.

Reply:
Nevermind, I realize that doesn't work.